web/js/MapUI/Components/UpdateProjection/UpdateProjection.js

```diff
@@ -0,0 +1,399 @@
+      case compareConstants.CHANGE_STATE:
+        if (compareMode === 'spy') {
+          return reloadLayers();
+        }
+        return;
+      case layerConstants.REORDER_LAYERS:
+      case layerConstants.REORDER_OVERLAY_GROUPS:
+      case compareConstants.TOGGLE_ON_OFF:
+      case compareConstants.CHANGE_MODE:
+        reloadLayers();
+        preloadForCompareMode();
+        return;
+      case layerConstants.TOGGLE_OVERLAY_GROUPS:
+        return reloadLayers();
+      case paletteConstants.SET_THRESHOLD_RANGE_AND_SQUASH:
+      case paletteConstants.SET_CUSTOM:
+      case paletteConstants.SET_DISABLED_CLASSIFICATION:
+      case paletteConstants.CLEAR_CUSTOM:
+      case layerConstants.ADD_LAYERS_FOR_EVENT:
+        return setTimeout(reloadLayers, 100);
+      case vectorStyleConstants.SET_FILTER_RANGE:
+      case vectorStyleConstants.SET_VECTORSTYLE:
+      case vectorStyleConstants.CLEAR_VECTORSTYLE:
+      case SET_SCREEN_INFO:
+        return onResize();
+      default:
+        break;
+    }
+  };
+
+  const flyToNewExtent = function(extent, rotation) {
+    const coordinateX = extent[0] + (extent[2] - extent[0]) / 2;
+    const coordinateY = extent[1] + (extent[3] - extent[1]) / 2;
+    const coordinates = [coordinateX, coordinateY];
+    const resolution = ui.selected.getView().getResolutionForExtent(extent);
+    const zoom = ui.selected.getView().getZoomForResolution(resolution);
+    // Animate to extent, zoom & rotate:
+    // Don't animate when an event is selected (Event selection already animates)
+    return fly(ui.selected, proj, coordinates, zoom, rotation);
+  };
+
+  /**
+ * Create a Layergroup given the date and layerGroups
+ */
+  async function getCompareLayerGroup([compareActiveString, compareDateString], state, granuleOptions) {
+    const { createLayer } = ui;
+    const compareSideLayers = getActiveLayers(state, compareActiveString);
+    const layers = getLayers(state, { reverse: true }, compareSideLayers)
+      .map(async (def) => {
+        const options = {
+          ...getGranuleOptions(state, def, compareActiveString, granuleOptions),
+          date: getSelectedDate(dateCompareState, compareDateString),
+          group: compareActiveString,
+        };
+        return createLayer(def, options);
+      });
+    const compareLayerGroup = await Promise.all(layers);
+
+    return new OlLayerGroup({
+      layers: compareLayerGroup,
+      date: getSelectedDate(dateCompareState, compareDateString),
+      group: compareActiveString,
+    });
+  }
+
+  /**
+  * Remove Layers from map
+  *
+  * @method clearLayers
+  * @static
+  *
+  * @param {object} map - Openlayers Map obj
+  *
+  * @returns {void}
+  */
+  const clearLayers = function() {
+    const activeLayers = ui.selected
+      .getLayers()
+      .getArray()
+      .slice(0);
+    lodashEach(activeLayers, (mapLayer) => {
+      ui.selected.removeLayer(mapLayer);
+    });
+    ui.cache.clear();
+  };
+
+  /**
+   * @method reloadLayers
```

Comment:
The params listed in this definition don't reflect the args used.

Reply:
I removed the `map` param description since it was not being used. It looks like `granuleDates` and `id` are destructured from `granuleOptions` which is why I believe they are listed with an indentation, so I am going to leave those.